core/state/database.go

```diff
@@ -307,26 +324,28 @@ func (fdb *ForkingDB) EndTransition() {
 	  |____|  |___|  /\___        \___  |____/\___  |   __/|___|  (____  |___|  |__|      |___|  (____  /_____/     |____(____  |___|  \____ |\___  \____ |
                                                     |__|`)
 	fdb.ended = true
+	if err := fdb.VerkleDB.diskdb.Put(dbKeyOverlayTransitionEnded, []byte{1}); err != nil {
+		return fmt.Errorf("failed to mark transition as ended: %s", err)
+	}
+	return nil
 }
 
 func (fdb *ForkingDB) AddTranslation(orig, trans common.Hash) {
 	// TODO make this persistent
 	fdb.translatedRootsLock.Lock()
 	defer fdb.translatedRootsLock.Unlock()
-	fdb.translatedRoots[fdb.translationIndex] = trans
-	fdb.origRoots[fdb.translationIndex] = orig
-	fdb.translationIndex = (fdb.translationIndex + 1) % len(fdb.translatedRoots)
+	fdb.diskdb.Put(append(translatedRootPrefix, orig.Bytes()...), trans.Bytes())
+	log.Info("Saved translation", "originRoot", orig.Bytes(), "translatedRoot", trans.Bytes())
```

Comment:
Instead of writing to the map, we save into the database.
I left an extra log-info since it can help a bit after when something doesn't work.